server/queryHandlers/ScanInstancesScheduleStatus.js

```diff
@@ -0,0 +1,131 @@
+/* Copyright (c) Trainline Limited, 2016. All rights reserved. See LICENSE.txt in the project root for license information. */
+'use strict';
+
+let resourceProvider = require('modules/resourceProvider');
+let co = require('co');
+let _ = require('lodash');
+let sender = require('modules/sender');
+let config = require('config');
+let scheduling = require('modules/scheduling');
+
+module.exports = function ScanInstancesScheduleStatusQueryHandler(query) {
+  return co(function*() {
+
+    let instances = yield getInstances(query);
+    let scheduledActions = scheduledActionsForInstances(instances);
+
+    return scheduledActions;
+
+  });  
+};
+
+function getInstances(query) {
+
+  return Promise.all([getAllInstances(query), getAllEnvironments(query), getAllASGs(query)]).then(data => {
+
+    let allInstances = data[0];
+    let environments = buildEnvironmentIndex(data[1]);
+    let asgData = data[2];
+    let asgs = buildASGIndex(data[2]);
+
+    let instances = [];
+
+    allInstances.forEach(instance => {
+      let environmentName = getInstanceTagValue(instance, 'environment');
+      instance.Environment = findInIndex(environments, environmentName);
+
+      let asgName = getInstanceTagValue(instance, 'aws:autoscaling:groupName');
+      instance.AutoScalingGroup = findInIndex(asgs, asgName);
+
+      instances.push(instance);
+    });
+
+    return instances;
+
+  });
+
+}
+
+function scheduledActionsForInstances(instances) {
+
+  let dateTime = new Date();
+
+  return instances.map(instance => {
+    let action = scheduling.actionForInstance(instance, dateTime);
+    return { action, instance: instance.InstanceId };
+  });
+
+}
+
+function buildEnvironmentIndex(environmentData) {
```

Comment:
are you sure about these Environment / Asg indexes? Why not keep array and just use Lodash `_.find` instead of `findInIndex`? you can do ```_.find(Environments, (environment) => environment.Value.EnvironmentName === 'myenv')```